jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java

```diff
@@ -5990,15 +5990,8 @@ protected void applyTaskProperties(Task task,
                     Assignment a = Bpmn2Factory.eINSTANCE.createAssignment();
                     FormalExpression fromExpression = Bpmn2Factory.eINSTANCE.createFormalExpression();
                     if (assignmentParts.length > 1) {
-                        // check if custom workitem property has been updated
-                        if (isCustomElement(properties.get("tasktype"),
-                                            preProcessingData) &&
-                                (properties.get(fromPart.toLowerCase()) != null && properties.get(fromPart.toLowerCase()).length() > 0)) {
-                            fromExpression.setBody(properties.get(fromPart.toLowerCase()));
-                        } else {
-                            String replacer = decodeAssociationValue(assignmentParts[1]);
-                            fromExpression.setBody(wrapInCDATABlock(replacer));
-                        }
+                        String replacer = decodeAssociationValue(assignmentParts[1]);
```

Comment:
The change I've added to propertywindow.js (and function definitions in dataioeditor.js) ensure that it isn't broken. When the user sets a custom property, ORYX.DataIOEditorUtils.setAssignmentsPropertyForCustomAssignment is called which updates the 'assignments' property with the new custom property value setting. When the BP is serialized, the values for custom properties are taken from the 'assignments' property rather than from the custom properties themselves.